src/ray/core_worker/core_worker.cc

```diff
@@ -238,6 +248,24 @@ void CoreWorkerProcess::RunTaskExecutionLoop() {
   instance_.reset();
 }
 
+void CoreWorkerProcess::RunStatsService() {
+  absl::MutexLock lock(&CoreWorkerProcess::stats_mutex_);
+  if (CoreWorkerProcess::stats_thread_) {
+    return;
+  }
+  // Initialize stats.
+  const ray::stats::TagsType global_tags = {{ray::stats::ComponentKey, "core_worker"},
+                                            {ray::stats::VersionKey, "0.9.0.dev0"}};
+
+  ray::stats::Init(global_tags, RayConfig::instance().metrics_agent_port(),
+                   CoreWorkerProcess::stats_io_service_);
+  CoreWorkerProcess::stats_thread_.reset(
+      new std::thread([]() { CoreWorkerProcess::stats_io_service_.run(); }));
+  // TODO(lingxuan.zlx): Should shutdown if opencensus disabled.
+  // Detach to avoid crash when process exit.
+  CoreWorkerProcess::stats_thread_->detach();
```

Comment:
Isn't the lifecycle of stats going together with processes? Why do we need to avoid crash?

Reply:
> Isn't the lifecycle of stats going together with processes? Why do we need to avoid crash?

It's crashed in after core_worker_test finished and return abnormal value, which makes gtest failed.